### DISCUSSION TITLE
Windows libpsl equivalent

### DISCUSSION THREAD

**Discussion by BrianInglis** [curl/curl#7114] [Q&A] [answered]
Was asked a question about why Cygwin does not provide Mingw `libpsl` libraries, so I assume MS Windows natively provides some equivalent info, but can not find any obvious references to it by searching, in `curl` or `wget`/2, so it may be buried somewhere in Microsoft-ese terminology. 
What is the MS Windows equivalent of `libpsl`, or is that info provided or handled some other way?

**Comment by bagder** [comment, marked as answer]
I'm not an authority on the Public Suffix nor Windows native APIs but I've never heard about any such native API. I don't think one exists.

**Comment by BrianInglis** [comment]
It looks like `curl` and `wget`/2 dummy out `psl` access on MS Windows. 
IE seems to have a list res://urlmon.dll/ietldlist.xml, with suffix issues in IE and also Outlook, but that may be because the list may support only TLDs not suffixes.
Outlook may have no such list, even long time TLDs are not supported or handled properly, it seems to be up to the user to specify the correct values in rules.

**Reply by bagder** [reply]
curl doesn't care about platform for PSL. It will use libpsl if present, independent of platform.
